mlir/lib/Target/LLVMIR/DebugTranslation.cpp

```diff
@@ -330,7 +330,10 @@ llvm::DILocation *DebugTranslation::translateLoc(Location loc,
   if (auto callLoc = dyn_cast<CallSiteLoc>(loc)) {
     // For callsites, the caller is fed as the inlinedAt for the callee.
     auto *callerLoc = translateLoc(callLoc.getCaller(), scope, inlinedAt);
-    llvmLoc = translateLoc(callLoc.getCallee(), scope, callerLoc);
+    llvmLoc = translateLoc(callLoc.getCallee(), nullptr, callerLoc);
```

Comment:
Hi, I'm working with triton and I'm not sure I fully understand this change - is this scope->nullptr really intentional? It looks like in no place you are propagating the scope? We've found that while we previously were getting the correct inlined location propagated for our debugging, we no longer are - now it propagates the parent scope instead of the inlined function location (ex: https://github.com/openai/triton/blob/main/python/test/unit/language/test_line_info.py#L143, we expect it to correctly be able to return the inlined function location, line 22, but it's now returning line 30, where the inlined function is called). Is it no longer supported that we see the location within the inlined function or is this a bug?

Reply:
Hi, thanks for checking, yes the nullptr here is intentional so that we don't push the parent scope into the callee scope (in case the callee didn't have a scope, and accidentally takes on the caller scope). This is actually to service the semantics change introduced here, which is that debug locations of callees are annotated in the callee location, not the overall callsite location (following the motivation mentioned in the PR description).

For example, what previously might look like
```
fused<#callee>[callsite("foo.mlir":2:4 at fused<#sp0>["foo.mlir":28:5])]
```
should now look like
```
callsite(fused<#callee>["foo.mlir":2:4] at fused<#sp0>["foo.mlir":28:5])
```

Can you double check that your llvm dialect IR is generated as expected?

Reply:
Thank you :) I was able to  change it to reflect this